test-data/unit/check-enum.test

```diff
@@ -2125,3 +2125,32 @@ class AllPartialList(Enum):
 
     def check(self) -> None:
         reveal_type(self.value)  # N: Revealed type is "builtins.list[Any]"
+
+[case testEnumLiteralsSupportedByPlugin]
+from enum import Enum
+from typing_extensions import Literal
+
+class E(Enum):
+    A = 1
+    B = 2
+
+x: Literal[E.A]
+reveal_type(x)  # N: Revealed type is "Literal[__main__.E.A]"
+reveal_type(x.value)  # N: Revealed type is "Literal[1]?"
+reveal_type(x.name)  # N: Revealed type is "Literal['A']?"
+
+e: E
+if e is E.A:
+    reveal_type(e)  # N: Revealed type is "Literal[__main__.E.A]"
+    reveal_type(e.value)  # N: Revealed type is "Literal[1]?"
+    reveal_type(e.name)  # N: Revealed type is "Literal['A']?"
+else:
+    reveal_type(e)  # N: Revealed type is "Literal[__main__.E.B]"
+    reveal_type(e.value)  # N: Revealed type is "Literal[2]?"
+    reveal_type(e.name)  # N: Revealed type is "Literal['B']?"
+
+y: Literal[E.A, E.B]
+reveal_type(y)  # N: Revealed type is "Union[Literal[__main__.E.A], Literal[__main__.E.B]]"
+reveal_type(y.value)  # N: Revealed type is "Any"
```

Comment:
At this point it's fair that it's `Any` rather than `int`?

Reply:
IIRC I fix this in my other PR about enums. Given that, I think this is alright? I won't resolve this until I actually check that other PR fixes this, of course.

Reply:
Nevermind, the other PR doesn't fix this. I'll fix it.

Reply:
Turns out fixing this fixes `.value` on unions of enums!